app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventEditors.kt

```diff
@@ -0,0 +1,268 @@
+                    negativeButton(text = stringResource(R.string.cancel)) {
+                        dateDialogState.hide()
+                        onAction(CancelEditMode)
+                    }
+                }
+            ) {
+                datepicker(
+                    initialDate = editMode.dateTime.toLocalDate(),
+                    title = editMode.dialogTitle.get,
+                ) {
+                    pickedDate = it.atTime(editMode.dateTime.toLocalTime())
+                }
+            }
+        }
+        is EditMode.ChooseFromTime,
+        is EditMode.ChooseToTime -> {
+            editMode as EditMode.DateTimePayload
+            var pickedTime by remember { mutableStateOf(LocalDateTime.now()) }
+            val dateDialogState = rememberMaterialDialogState()
+
+            dateDialogState.show()
+            MaterialDialog(
+                dialogState = dateDialogState,
+                buttons = {
+                    positiveButton(text = stringResource(R.string.ok)) {
+                        onAction(EditMode.UpdateDateTime(pickedTime?.toZonedDateTime()!!))
+                    }
+                    negativeButton(text = stringResource(R.string.cancel)) {
+                        dateDialogState.hide()
+                        onAction(CancelEditMode)
+                    }
+                }
+            ) {
+                timepicker(
+                    initialTime = editMode.dateTime.toLocalTime(),
+                    title = editMode.dialogTitle.get,
+                ) {
+                    pickedTime = it.atDate(editMode.dateTime.toLocalDate())
+                }
+            }
+        }
+        is EditMode.ChooseRemindAtDateTime -> { // handled in the RemindAt UI element, this is here to remove compiler warning
+        }
+        is EditMode.ChooseAddPhoto -> TODO()
+        is EditMode.ConfirmRemovePhoto -> TODO()
+        is EditMode.ChooseAddAttendee -> {
+            val addAttendeeDialogState = rememberMaterialDialogState()
+            var attendeeEmail by remember { mutableStateOf("") }
+
+            addAttendeeDialogState.show()
+            MaterialDialog (
+                dialogState = addAttendeeDialogState,
+                properties = DialogProperties(
+                    dismissOnBackPress = true,
+                    dismissOnClickOutside = true
+                ),
+                buttons = {
+                    positiveButton(
+                        text = stringResource(R.string.ok),
+                        disableDismiss = true,
+                        textStyle = MaterialTheme.typography.button.copy(
+                            if(state.isAttendeeEmailValid == true)
+                                MaterialTheme.colors.onSurface
+                            else
+                                MaterialTheme.colors.onSurface.copy(alpha = 0.3f)
+                        )
+                    ) {
+                        if(state.isAttendeeEmailValid == true) {
```

Comment:
The comparison seems to not be necessary here (or is `isAttendeeEmailValid` nullable and if so why is it?)

Reply:
OK, i will check this.

Reply:
The reason it is necessary is because its null when the edit text field is blank. Once the user starts typing into the text field, it will be set to true or false and thats when I want to show the icon. 

I dont want to present the user with an error state before they even start typing. 

Does this make sense?

Reply:
Ah, you do live validation. I thought this is just about whether the checkmark is displayed or not in the email field. In my app I only display an actual error message if the user tries to click on add when the email is invalid. But if you do it differently, it's fine

Reply:
OK sounds good.